PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,289 @@
+     *
+     * @param holder The holder for the problems found
+     * @return The visitor for the inspection
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new SingleOperationInLoopVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * Visitor class to traverse the PSI tree and check for single Azure client operation inside a loop.
+     * The visitor will check for loops of type for, foreach, while, and do-while.
+     * The visitor will check for a Text Analytics client operation inside the loop.
+     * If a Text Analytics client operation is found inside the loop, and the API has a batch alternative, a problem will be registered.
+     */
+    static class SingleOperationInLoopVisitor extends JavaElementVisitor {
+
+        // Define the holder for the problems found and whether the inspection is running on the fly
+        private final ProblemsHolder holder;
+
+        // // Define constants for string literals
+        private static final RuleConfig ruleConfig;
+        private static final String SUGGESTION;
+        private static final List<String> AVAILABLE_BATCH_METHODS;
+        private static boolean SKIP_WHOLE_RULE;
+
+        static {
+            final String ruleName = "SingleOperationInLoopCheck";
+            RuleConfigLoader centralRuleConfigLoader = RuleConfigLoader.getInstance();
+
+            // Get the RuleConfig object for the rule
+            ruleConfig = centralRuleConfigLoader.getRuleConfig(ruleName);
+
+            AVAILABLE_BATCH_METHODS = ruleConfig.getMethodsToCheck();
+            SUGGESTION = ruleConfig.getAntiPatternMessage();
+            SKIP_WHOLE_RULE = ruleConfig == RuleConfig.EMPTY_RULE || AVAILABLE_BATCH_METHODS.isEmpty();
+        }
+
+        /**
+         * Constructor for the visitor
+         *
+         * @param holder     The holder for the problems found
+         * @param isOnTheFly Whether the inspection is running on the fly. If true, the inspection is running as you type.
+         */
+        public SingleOperationInLoopVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+        }
+
+        /**
+         * Visit the for statement and check for single Azure client operation inside the loop.
+         *
+         * @param statement The for statement to check
+         */
+        @Override
+        public void visitForStatement(@NotNull PsiForStatement statement) {
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the foreach statement and check for single Azure client operation inside the loop.
+         *
+         * @param statement The foreach statement to check
+         */
+        @Override
+        public void visitForeachStatement(@NotNull PsiForeachStatement statement) {
```

Comment:
```suggestion
        public void visitForEachStatement(@NotNull PsiForeachStatement statement) {
```

Reply:
this is a default method name from the JavaElementVisitor so I can't change it